pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -1608,8 +1609,8 @@ func (pl *dynamicResources) reserveClaim(ctx context.Context, state *stateData,
 	if allocationPatch != "" {
 		// The scheduler was handling allocation. Now that has
 		// completed, either successfully or with a failure.
-		if err != nil {
-			pl.claimAssumeCache.Restore(claim.Namespace + "/" + claim.Name)
+		if err == nil {
+			pl.claimAssumeCache.Assume(claim)
```

Comment:
what if the scheduler observes the patch update and updates the "assumeCache" before this line runs?

Maybe it's still better to call `Assume` before calling `Patch`?

Reply:
> what if the scheduler observes the patch update and updates the "assumeCache" before this line runs?

`Assume` checks that it is not overwriting a newer object. If they are the same, then it makes no difference.

> Maybe it's still better to call Assume before calling Patch?

That was the mistake that I made earlier: `Assume` must only be called with an object that has been received from the apiserver. Otherwise the resource version is not correct and the "newer than" comparison breaks down.

Reply:
Ok, then the only other scenario would be if the Claim is deleted before `Assume` runs, but we have the finalizer to protect it.

Reply:
The `Assume` implementation also handles concurrent deletion: it only updates an existing object, but doesn't allow "creating" it.

But you are right, in this context we are (almost) guaranteed to never encounter that scenario.